soda_machine.rb

```diff
@@ -1,18 +1,40 @@
 class SodaMachine
-  attr_reader :sodas, :cash
+  attr_accessor :sodas, :cash
 
   def initialize(args = {})
     @sodas = args[:sodas]
     @cash = args[:cash]
   end
 
   def current_inventory_count
+    #return number of sodas it contains
+    @sodas.length
   end
 
   def find_soda(soda_brand)
+    #find a single soda with a given brand
+    #iterate through sodas and when soda = brand return
+    @sodas.each do |cur_soda|
+      if cur_soda.brand == soda_brand 
+        return cur_soda
+        else
+          return nil
+      end
+    end
+
   end
 
   def sell(soda_brand)
```

Comment:
Great! Let's refactor :) why not check that the soda exists to sell first then do the buying & deleting? 

```
  def sell(soda_brand)
    return nil if find_soda(soda_brand) == nil
    soda_to_sell = find_soda(soda_brand)
    @cash += soda_to_sell.price
    @sodas.delete(soda_to_sell)
  end
```